acceptance/tests/custom_facts/windows_not_expand_command.rb

```diff
@@ -0,0 +1,34 @@
+test_name 'FACT-2054: Custom facts that execute on windows should raise an error' do
+  tag 'risk:low'
+
+  confine :to, :platform => /windows/
+
+  content = <<-EOM
+    Facter.add(:foo) do
+      setcode do
+        Facter::Core::Execution.execute("cmd", {:expand => false})
+      end
+    end
+  EOM
+
+  agents.each do |agent|
+    fact_dir = agent.tmpdir('facter')
+    fact_file = File.join(fact_dir, 'test_facts.rb')
+    create_remote_file(agent, fact_file, content)
+    env = {'FACTERLIB' => fact_dir}
+
+    teardown do
+      on(agent, "rm -rf '#{fact_dir}'")
+    end
+
+    step "Agent: Verify that exception is raised" do
+      begin
+        on(agent, facter('foo', :environment => env))
+      rescue Beaker::Host::CommandFailure => ex
+        assert(ex.message.include?("Unsupported argument on Windows expand"))
+      ensure
+        assert(ex != nil, "I should always raise an error")
```

Comment:
instead of a begin..rescue you can pass `:acceptable_exit_codes = [0, 1]` to `on`, then assert the output of stderr

i.e.

```ruby
on(agent, facter('foo', :environment => env), :acceptable_exit_codes => [0, 1]) do |output|
  assert_match(/Unsupported argument on Windows expand/, output.stderr, 'expand => false on Windows should raise')
end
```

Reply:
excelent